src/main/java/julia/git/GitObject.java

```diff
@@ -0,0 +1,34 @@
+package julia.git;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public interface GitObject {
+    String getContent();
+
+    default String getHash() {
+        String content = getContent();
+        StringBuilder result = new StringBuilder();
+        byte[] sha = sha1(content);
+        for (byte b : sha) {
+            int decimal = (int) b & 0xff;
+            String hex = Integer.toHexString(decimal);
+            if (hex.length() % 2 == 1) {
+                hex = "0" + hex;
+            }
+            result.append(hex);
+        }
+        return result.toString();
+    }
+
+    default byte[] sha1(String content) {
```

Comment:
This shouldn't be part of the interface. None of implementations need it. It should be in some other constant class and the method can be static.